cpp/src/arrow/record_batch.h

```diff
@@ -82,10 +82,15 @@ class ARROW_EXPORT RecordBatch {
   /// \brief Construct record batch from struct array
   ///
   /// This constructs a record batch using the child arrays of the given
-  /// array, which must be a struct array.  Note that the struct array's own
-  /// null bitmap is not reflected in the resulting record batch.
+  /// array, which must be a struct array.
+  ///
+  /// \param[in] pool the memory pool to allocate new validity bitmaps
```

Comment:
Because you added the `pool` param documentation, it now seems to complain that `array` is not documented